### ISSUE TITLE
Run a command before highstate but after copying files

### ISSUE THREAD

**Issue by daks** [saltstack/kitchen-salt#295]
Hello,

we are successfully using kitchen-salt for our infrastructure code since a long time, and are now trying to setup a new process _(inspired by librarian-{puppet,chef})_ to use formulas :
* define a file with all our dependencies/formulas
* run a command to install those in a defined directory
* run highstate

I first used `init_environment` to do it with success, but I want my 'init command' to be a shell script _(so I can have options and better error-handling)_ and `init_environment` seems to run before copying files.

Is there any other directive to do it? Do you think it could be added to `kitchen-salt` as I think it can be a common need.

**Comment by kiemlicz** [comment]
I have exactly the same question

`init_environment` runs too soon for me, I have to run a bunch of commands that already require State Tree to be present on the filesystem + salt installed

**Comment by gtmanfred** [comment]
Have you tried `install_after_init_environment` set to `True`?

https://github.com/saltstack/kitchen-salt/blob/master/docs/provisioner_options.md#install_after_init_environment

**Comment by kiemlicz** [comment]
I want to run some commands **after** the Salt is installed and **before** the `highstate` runs
With `install_after_init_environment` I don't get Salt installed

It is crucial to run some commands **after** Salt install **and** State tree copy due to https://github.com/saltstack/salt/issues/55437

**Comment by daks** [comment]
@gtmanfred I thought it would solve my problem (even if not all) but it's not the case. 
```
  install_after_init_environment: true
  init_environment: "/tmp/kitchen/srv/salt/bin/install_formulas.sh"
```
fails because it still tries to run the command *before* copying files.

**Comment by daks** [comment]
Closed after merge of #297
